UI/homepage.py

```diff
@@ -101,28 +235,65 @@ def handle_ui_events(event):
                 draw_filter_search_screen()
             elif event.ui_element == button_recipe:
                 draw_recipe_search_screen()
-            elif back_button and event.ui_element == back_button:
+            elif event.ui_element == back_button:
                 title.show()
                 button_filter.show()
                 button_recipe.show()
                 back_button.hide()
-                input_box.hide()
-                filter_title.hide()
                 if recipe_title:
                     recipe_title.hide()
-                search_results.hide()
+                if recipe_search_button:
+                    recipe_search_button.hide()
+                for checkbox in check_boxes:
+                    checkbox.checked = False
+                for filter_button in filter_buttons:
+                    filter_button.hide()
                 current_screen = 'home'
+
+                if input_box:
+                    input_box.hide()
+                item_list = []
```

Comment:
if this is to initialize `item_list` as a list, why not do it further down when it's actually called? Also it's global so it would probably work but scope-wise but this is in the if-segment of this big if statement but it's only called and utilized in the elif part of the if statement